t/loss.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2017 Fastly, Kazuho Oku
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#include "quicly/loss.h"
+#include "quicly/defaults.h"
+#include "test.h"
+
+#define ASSERT(expr)                                                                                                               \
+    do {                                                                                                                           \
+        int ret = !!(expr);                                                                                                        \
+        assert(ret);                                                                                                               \
```

Comment:
I've never seen this idiom... why isn't `assert(expr);` alone simply enough?

Reply:
Ah, to make sure that the expression is certainly evaluated even with `-DNDEBUG`?

Reply:
That's correct.

OTOH, I do not think it is a good idea to rely on `assert` for confirming the behavior. One reason might be to reduce the amount of test output, but we aren't making that much of calls in this file. Hence switching to using `ok` in c72f219.